### DISCUSSION TITLE
usage of rtsp_transport option for RTSP paths

### DISCUSSION THREAD

**Discussion by fabianeichfeldt** [bluenviron/mediamtx#3545] [Questions]
### Question

Hey there,

I am using MediaMTX as a proxy between my cameras and my streaming server. Therefor I am controlling MediaMTX via ControlApi and add/delete/modify paths on the fly.

But recently I noticed something which is not really working for me. Sometimes I want explicitly that MediaMTX connects via UDP to my camera. But even if I set `rtspTransport` to `udp` (int the path configuration) it eventually falls back to TCP in case UDP does not work. It just tries a bit longer (no scientific measurement taken, just feeling) to connect via UDP.
Is this behavior intended?

As a workaround I thought I can read somehow the rtsp connections/sessions via Control API and see there that it was fallen back to TCP. But this is also not possible, I can only see how my clients are connected to MediaMTX, but not how MediaMTX is connected to my camera.

Does anyone have an idea how to deal with it?
Thanks a lot in advance!

**Comment by fabianeichfeldt** [comment]
Can someone help me with this issue?
I basically just want to guarantee that a client can only connect via UDP/TCP, and no fallback should happen.

**Comment by aler9** [comment]
Hello, this was a bug that was fixed with https://github.com/bluenviron/gortsplib/pull/607

[a comment by an automated account is not shown]
